### ISSUE TITLE
C4 Owner Bug

### ISSUE THREAD

**Issue by Maxi605** [GuskiS/Trouble-in-Terrorist-Town#13]
There's a bug with the C4 when somebody else planted a C4 while you change your C4 Timer (I THINK IT IS LIKE THIS, MAYBE NOT) he's C4 is 'yours'.

Logs:

[LOG] 06/28/2018 - 19:494: [ITEM ] --- Player La Rana Super Manca xD! bought item Bomba C4 with ID 0
[LOG] 06/28/2018 - 19:49:44: [KILL ] --- [Inocente] Luzbelito;) [AF] was killed by [Traidor] Rivotril (V23B) with xm1014
[LOG] 06/28/2018 - 19:49:44: [DAMAGE ] --- [Traidor] Rivotril (V23B) attacked [Inocente] Luzbelito;) [AF] with 127 damage
[LOG] 06/28/2018 - 19:49:49: [ITEM ] --- Player Rivotril (V23B) bought item Bomba C4 with ID 0
[LOG] 06/28/2018 - 19:49:49: [MISC ] --- Dead body for Luzbelito;) [AF] was created, ent: 155
[LOG] 06/28/2018 - 19:49:51: [ITEM ] --- Rivotril (V23B) planted the C4
[LOG] 06/28/2018 - 19:49:54: [ITEM ] --- Rivotril (V23B) planted the C4

**Comment by GuskiS** [comment]
But were there 2 bombs planted?

**Comment by Maxi605** [comment]
No, i planted a C4 at the same exact time another player did, so, it cosidered that both C4's are mine

**Comment by Maxi605** [comment]
The bug happen again,
This is the entire round, as you can see two different players bought a C4 but only one of them planted 2 C4's (When he never bought 2)

The log show really bad, my bad, here's the log itself
[TTT20180807.log](https://github.com/GuskiS/Trouble-in-Terrorist-Town/files/2268505/TTT20180807.log)

**Comment by GuskiS** [comment]
Hmm. It is shame that it doesn't show the second C4 exploding - that would allow us to speculate if owner is truly wrong or just the logging part is incorrect.

Basicly, owner would receive karma for the bombing. If all karma would go to 1 person, then it is bug with the owner, however, if the karma is given to both, then it is something with logging.

**Comment by Maxi605** [comment]
Ingame shows 2 C4's loading to explode by the same exact owner.

**Comment by GuskiS** [comment]
https://github.com/GuskiS/Trouble-in-Terrorist-Town/blob/7a0788888d708ef17914a0b04f569c7ec5a2cfe5/scripting/ttt_item_c4.sma#L236
on this line add:
```
log_amx("C4 bomb_planted: %d == %d == %d == %d", c4, id, entity_get_int(c4, EV_INT_iuser1), entity_get_edict(c4, EV_ENT_owner));
```

https://github.com/GuskiS/Trouble-in-Terrorist-Town/blob/7a0788888d708ef17914a0b04f569c7ec5a2cfe5/scripting/ttt_item_c4.sma#L265
before this line, add:
```
log_amx("C4 set_c4_info: %d == %d == %d == %d", c4, id, entity_get_int(c4, EV_INT_iuser1), entity_get_edict(c4, EV_ENT_owner));
```

and let me know those when this issue happens.
